nautilus_core/common/src/lib.rs

```diff
@@ -35,3 +37,12 @@ pub mod python;
 
 #[cfg(feature = "redis")]
 pub mod redis;
+
+pub trait LogWriter {
+    /// Writes a log line.
```

Comment:
Perhaps putting this in a separate module makes more sense. The implementation of the specific loggers stdout, stderr etc can also go in the same module.

This will also keep the logging module focused on the logging infrastructure rather than the IO part of it.